### PR TITLE
Add WebhookNotify plugin

### PR DESCRIPTION
**Pull request by leewillis77**
## Contribution type
New plugin

## Description of change
Adds a webhook notify that sends a JSON webhook to a specified URL. Includes plugin, plus usage documentation.

### CODE REVIEW

File: docs/en/plugins/webhook_notify.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+|-------|-----------|-------------|
+| `url` | Yes | The URL to send the webhook to |
+
+### Examples
+
+Send a message if the build fails:
+
+```yaml
+failure:
+    webook_notify:
```
Comment (corpsee) [inline docs/en/plugins/webhook_notify.md]: 'webook_notify' -> 'webhook_notify'

File: docs/en/plugins/webhook_notify.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        url: "http://example.com/webhook-handler"
+```
+
+Send a message if the build is successful:
+
+```yaml
+success:
+    webook_notify:
```
Comment (corpsee) [inline docs/en/plugins/webhook_notify.md]: 'webook_notify' -> 'webhook_notify'

File: docs/en/plugins/webhook_notify.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        url: "http://example.com/webhook-handler"
+```
+
+Send a message every time the build runs:
+
+```yaml
+complete:
+    webook_notify:
```
Comment (corpsee) [inline docs/en/plugins/webhook_notify.md]: 'webook_notify' -> 'webhook_notify'

File: src/Plugin/WebhookNotify.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+use PHPCensor\Plugin;
+
+/**
+ * Webhook notify Plugin
+ *
+ * @author Lee Willis (Ademti Software) : https://www.ademti-software.co.uk
+ */
+class WebhookNotify extends Plugin
+{
+    const STATUS_PENDING = 0;
```
Comment (corpsee) [inline src/Plugin/WebhookNotify.php:20]: This constants already exist in parent Plagin class.

File: src/Plugin/WebhookNotify.php [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            'commit_id'       => $this->build->getCommitId(),
+            'short_commit_id' => substr($this->build->getCommitId(), 0, 7),
+            'branch'          => $this->build->getBranch(),
+            'branch_link'     => $this->build->getBranchLink(),
+            'committer_email' => $this->build->getCommitterEmail(),
+            'commit_message'  => $this->build->getCommitMessage(),
+            'commit_link'     => $this->build->getCommitLink(),
+            'build_link'      => APP_URL . 'build/view/' . $this->build->getId(),
+            'project_link'    => APP_URL . 'project/view/' . $this->build->getProjectId(),
+            'statusCode'      => $this->build->getStatus(),
```
Comment (corpsee) [inline src/Plugin/WebhookNotify.php:78]: Could you do payload keys more consistent by name please ('example_key' or 'exampleKey' for all)?

**Comment by corpsee** [comment]
@leewillis77 It is a good job. Thanks!

**Comment by leewillis77** [comment]
All comments fixed - thanks for the quick review!

I'd like to put the correct version number in the user-agent string (see https://github.com/leewillis77/php-censor/blob/master/src/Plugin/WebhookNotify.php#L81) - but not sure how I can get to the Application class cleanly? (Campfire & HipChat also hardcode version 1.0 for presumably the same reason?)

If you can advise happy to update all three of those to have a dynamic version number.

**Comment by corpsee** [comment]
@leewillis77 You can use file VERSION.md. See for example: https://github.com/php-censor/php-censor/blob/master/src/Console/Application.php#L86.

Could you also fix version for Campfire & HipChat plugins please?

**Comment by leewillis77** [comment]
Done. Thanks

**Comment by corpsee** [comment]
@leewillis77 :+1:

**Comment by corpsee** [comment]
Merged to `master` and will be released with version `1.2.0`.

**Pull request merged**